plugins/sources-provider/src/test/java/eu/f4sten/sourcesprovider/utils/PayloadParsingTest.java

```diff
@@ -69,7 +68,7 @@ void findCPayloadTest() throws IOException, JSONException {
         var message = new JSONObject(Files.readString(getTestResource("PayloadParsingTest/c_metadatadb_extension_message.json").toPath()));
         var payload = pp.findSourcePayload(message);
         assertNotNull(payload);
-        assertEquals(Constants.debianForge, payload.getForge());
+        assertEquals("debian", payload.getForge());

```

Comment:
I'd prefer to have the `Constants` class instead of the hard-coded strings.

Reply:
I decided to get rid of the Constants class for three reasons:

- Dependency-wise, there is no elegant way right now that would allow sharing the Constants class between `data-processing` and `core`.
- I plan to phase out the other languages, as they are not maintained right now.
- I plan to refactor the `SourcesProvider` soon(tm) anyways, as I do not like that it extracts the jar contents.

I agree that magic strings are a *smell*, however, I see them as a temporary issue and they will vanish with further evolution of the code base anyways... Do you think I should still introduce a dedicated `FastenConstants` clone in the `data-processing` repo?

Reply:
I'd prefer that we introduce `FastenConstants` in the `data-processing` repo.
Also, I like to have the `SourcesProvider` plug-in in the repo. the JAR contents, i.e., Java source files, can be useful for training code language models.

Reply:
Yes and no. I would like to have the sources as well, but from a resource usage perspective, it does not make any sense to unpack the .jar files. I will store the source jars and users that need the contents can unpack them and reduce them to the bits they need.

I will create a `FastenConstants` class.

Reply:
Okay, it makes sense from the disk usage perspective.
I resolve the conversation.